packages/js/src/executors/release-publish/extract-npm-publish-json-data.ts

```diff
@@ -0,0 +1,62 @@
+const expectedNpmPublishJsonKeys = [
+  'id',
+  'name',
+  'version',
+  'size',
+  'filename',
+];
+
+export function extractNpmPublishJsonData(str: string): {
+  beforeJsonData: string;
+  jsonData: Record<string, unknown> | null;
+  afterJsonData: string;
+} {
+  const jsonRegex = /{(?:[^{}]|{[^{}]*})*}/g;
+  const jsonMatches = str.match(jsonRegex);
+  let jsonData: Record<string, unknown> | null = null;
+  let jsonStartIndex = -1;
+  let jsonEndIndex = -1;
+
+  if (jsonMatches) {
+    for (const match of jsonMatches) {
+      // Cheap upfront check to see if the stringified JSON data has the expected keys as substrings
+      if (!expectedNpmPublishJsonKeys.every((key) => str.includes(key))) {
+        continue;
+      }
+      // Full JSON parsing to identify the JSON object
+      try {
+        const parsedJson = JSON.parse(match);
+        if (
+          !expectedNpmPublishJsonKeys.every(
+            (key) => parsedJson[key] !== undefined
+          )
+        ) {
+          continue;
+        }
+        jsonData = parsedJson;
+        jsonStartIndex = str.indexOf(match);
+        jsonEndIndex = jsonStartIndex + match.length;
+        break;
+      } catch {
+        // Ignore parsing errors for unrelated JSON blocks
+      }
+    }
+  }
+
+  if (jsonData) {
+    const beforeJsonData = str.slice(0, jsonStartIndex);
+    const afterJsonData = str.slice(jsonEndIndex);
+    return {
+      beforeJsonData,
+      jsonData,
+      afterJsonData,
+    };
+  }
+
+  // No jsonData detected, the whole contents is the beforeJsonData
+  return {
+    beforeJsonData: str,
+    jsonData: null,
+    afterJsonData: '',
+  };
```

Comment:
I think we should throw an error here instead.

Reply:
I don’t think I agree, because we would then need dedicated handling within the existing try catch that this code executes within to get it to ignore this specific error only, it would currently blow up. That catch is already relatively involved for handling errors from npm and that would suddenly bloat its purpose to handle known “errors” from us as well. It’s either that or wrap it in another nested try catch which seems equally undesirable…

Reply:
Ultimately, it's subjective, with zero consequences to the user experience either way (assuming you agree with me that it shouldn't end up in the "something unexpected went wrong" section, so hopefully we don't need to block the PR over it